docs/pairing-architecture.md

```diff
@@ -0,0 +1,283 @@
+## Pairing Flow Architecture
+
+The Desktop and Mobile devices will not communicate directly, because we cannot assume that they will be on the same network and because this is difficult to achieve from web content. Instead, they will exchange encrypted messages via websocket to a lightweight "channel server" service which can relay messages back and forth:
+
+
+![](images/pairing-flow1.png)
+
+*****
+
+
+![](images/pairing-flow2.png)
+
+### Detailed Flow
+
+Is is advised to read the following section while looking at the detail digram while reading the information below.
+
+![](https://www.lucidchart.com/publicSegments/view/00b7d8a7-fd78-493f-91f5-120b4034f8b0/image.png)
+
+[View / Edit in LucidChart](https://www.lucidchart.com/documents/edit/1bc1b604-0047-4542-8827-ed8518b0433e/0)
+
+
+#### 1) Desktop opens an encrypted channel and advertises it via QR code
+
+To make itself available for pairing, the Desktop device opens a connection to the Channel Server using the [fxa-pairing-channel library](https://github.com/mozilla/fxa-pairing-channel). 
+Any application that learns the channel key will be able to request pairing from the Desktop device, or intercept messages exchanges during the pairing flow.
+This library uses a subset of TLS-like communication between the two devices.
+Once connected it will receive a server-allocated channel id, then the library will generate a 32-byte channel key to be used for client-side encryption of messages on the channel and derive a confirmation code that can be displayed to the user as visual confirmation the the devices are using the same key.
+ * The channel key is what secures the pairing flow.
+ * The channel id and channel key are encoded using base64url and then combined into a "pairing url". 
+ * The generated URL looks like: https://accounts.firefox.com/pair#channel_id=[channel_id]&channel_key=[channel_key]
+ * This URL is then encoded in a QR code and displayed on screen.
+ 
+#### 2) Mobile scans the QR code and connects to the encrypted channel
+
+To initiate pairing, the Mobile device scans the QR code displayed by Desktop and extracts the channel id and channel key.
+It opens a websocket to the Channel Server using the same fxa-pairing-channel library specifying the channel id, channel key.
+The Channel Server will now relay websocket messages from Mobile to Desktop and vice-versa.
```

Comment:
Is it worth adding a bit of extra info in here, that the logical for all this lives in the "supplicant broker" in this repo?